coalib/results/Result.py

```diff
@@ -70,7 +71,10 @@ def __init__(self,
         # Sorting is important for tuple comparison
         self.affected_code = tuple(sorted(affected_code))
         self.severity = severity
-        self.confidence = confidence
+        if confidence < 0 or confidence > 100:
+            raise ValueError('Value of confidence should be between 0 and 100.')
+        else:
+            self.confidence = confidence
```

Comment:
hey, normally you wouldn't do an else here, @Udayan12167 can you file a newcomer issue to remove the else?